src/SAML2/XML/saml/SubjectLocality.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\saml;
+
+use DOMElement;
+use SAML2\Utils;
+use Webmozart\Assert\Assert;
+
+/**
+ * Class representing SAML2 SubjectLocality
+ *
+ * @author Tim van Dijen, <tvdijen@gmail.com>
+ * @package simplesamlphp
+ */
+final class SubjectLocality extends AbstractSamlElement
+{
+    /** @var string|null */
+    protected $address;
+
+    /** @var string|null */
+    protected $dnsName;
+
+
+    /**
+     * Initialize an SubjectLocality.
+     *
+     * @param string|null $address
+     * @param string|null $dnsName
+     */
+    public function __construct(
+        ?string $address = null,
+        ?string $dnsName = null
+    ) {
+        $this->setAddress($address);
+        $this->setDnsName($dnsName);
+    }
+
+
+    /**
+     * Collect the value of the address-property
+     *
+     * @return string|null
+     */
+    public function getAddress(): ?string
+    {
+        return $this->address;
+    }
+
+
+    /**
+     * Set the value of the address-property
+     *
+     * @param string|null $address
+     * @return void
+     */
+    private function setAddress(?string $address): void
+    {
+        $this->address = $address;
```

Comment:
```php
        Assert::nullOrIp($address, 'Invalid IP address');
```

Reply:
I think we've had this discussion before in another place, but the schema says it's a `string`

Reply:
Yes, it is, but a string that represents an IP address. Maybe I'm misunderstanding `nullOrIp()`, but that's what I would expect it to test for.

Reply:
Don't you want to be liberal on the incoming end?

I remember adding similar restrictions on one of the mdui or mdrpi classes (v4) and then I had to undo it because people started to complain

Reply:
Here it is;  https://github.com/simplesamlphp/saml2/commit/1e67636360d7062b4e677cea9602a56198aceba3

Reply:
Yeah, in most occasions we should be flexible, but in others I'm not sure how much sense does it make to be so... In this case, the address is supposed to be exactly an IP address, so I don't really see the point of allowing anything else. What we could do in cases like this, though, is to continue but discarding the value.

Reply:
The other case was a bit more vague and described the value as 'network location'.. I think someone had a hostname in there...  I also remember a case of mdui iconUrl where the value was a base64-encoded image instead of a url..

Let's leave it for now

Reply:
Yes, those cases make full sense. One thing is validating input, and another different thing is discarding input that's actually legitimate. But in this case I think it should be fine to check for an IP.

In any case, not a big deal.